### PR TITLE
[SYCL-MLIR][NFC]: Move polygeist transform utils

### PR DESCRIPTION
**Pull request by etiotto**
Move the Loop Versioning utility classes and the Loop Guard utility classes into a utility file to be used for polygeist transformations.

### CODE REVIEW

File: polygeist/lib/Dialect/Polygeist/Transforms/Utils/LoopGuardBuilder.cpp
Comment (whitneywhtsang) [inline polygeist/lib/Dialect/Polygeist/Transforms/Utils/LoopGuardBuilder.cpp:1]: delete empty file.

File: polygeist/include/mlir/Dialect/Polygeist/Transforms/Utils/Utils.h
Comment (whitneywhtsang) [inline polygeist/include/mlir/Dialect/Polygeist/Transforms/Utils/Utils.h:1]: I see other Utils.h are directly under Transforms, is there a reason why you want to put it under Utils?
```
$ find . -name Utils.h | grep Transforms
./polygeist/lib/Dialect/Polygeist/Transforms/Utils.h
./mlir/include/mlir/Dialect/GPU/Transforms/Utils.h
./llvm/include/llvm/Transforms/Utils.h
./mlir-sycl/lib/Transforms/Utils.h
```

Comment (etiotto) [reply polygeist/include/mlir/Dialect/Polygeist/Transforms/Utils/Utils.h:1]: MLIR isn't consistent, for example we also have: 
```
./mlir/Dialect/Arith/Utils/Utils.h  (This header file defines prototypes for various transformation utilities for
// the Arith dialect....)
./mlir/Dialect/Linalg/Utils/Utils.h
```
Which places the Utils.h file under a `Utils` directory (although IMO it should be under the Transform directory given that the file claims to contain transformation utilities as per comment above).

I'd like to have a folder to contain the utilities. We can argue whether it should be a folder inside the `Transform` directory (my preference) or not. What do you prefer ?

Comment (etiotto) [reply polygeist/include/mlir/Dialect/Polygeist/Transforms/Utils/Utils.h:1]: Also the reason to have a folder is that I'd like to further split the utilities in their own implementation files in the future.

**Review by whitneywhtsang — APPROVED** [review]

File: polygeist/include/mlir/Dialect/Polygeist/Transforms/Utils/Utils.h
Comment (etiotto) [reply polygeist/include/mlir/Dialect/Polygeist/Transforms/Utils/Utils.h:1]: I decided to follow MLIR scheme of placing the `Utils` directory not under the `Transform` directory and I renamed the file to TransformUtils.[h,cpp].

**Review by whitneywhtsang — APPROVED** [review]

**Review by victor-eds — APPROVED** [review]

**Pull request merged**